classic_ml/LinearRegression/run_optimized.py

```diff
@@ -0,0 +1,107 @@
+
+import argparse
+import numpy as np
+from sklearn.linear_model._base import safe_sparse_dot
+from utils.benchmark import run_model
+from classic_ml.tabular_dataset import TabularDataset
+from utils.misc import print_goodbye_message_and_die, download_ampere_imagenet
+
+from sklearn.linear_model import  LinearRegression
+
+from sklearn.linear_model import  LinearRegression as LinearRegressionBase
+class LinearRegression(LinearRegressionBase):
+    #overwriting decision func from original scikit learn and in this it skips checking and validation
+    def _decision_function(self, X):
+        return safe_sparse_dot(X, self.coef_.T, dense_output=True) + self.intercept_
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Run DecisionTreeClassifier model.")
+    parser.add_argument("-m", "--model_path",
+                        type=str,
+                        help="path to the model")
+    parser.add_argument("-p", "--precision",
+                        type=str, choices=["fp32"], required=True,
+                        help="precision of the model provided")
+    parser.add_argument("-b", "--batch_size",
+                        type=int, default=1,
+                        help="batch size to feed the model with")
+    parser.add_argument("-f", "--framework",
+                        type=str,
+                        choices=["ort","sklearn"], required=True,
+                        help="specify the framework in which a model should be run")
+    parser.add_argument("--timeout",
+                        type=float, default=60.0,
+                        help="timeout in seconds")
+    parser.add_argument("--num_runs",
+                        type=int,
+                        help="number of passes through network to execute")
+    parser.add_argument("--data_path",
+                        type=str,
+                        help="path to csv file containing input data")
+    args = parser.parse_args()
+   
+    if args.framework != "sklearn" and args.model_path is None:
+        parser.error(f"You need to specify the model path when using {args.framework} framework.")
+    return args
+
+def run_ort_fp(model_path, batch_size, num_runs, timeout, data_path):
+    from utils.ort import OrtRunner
+
+    def run_single_pass(ort_runner, dataset):
+        X, y = next(dataset)
+        
+        ort_runner.set_input_tensor("input_tensor", X)
+        y_hat = ort_runner.run()        
+        dataset.submit_predictions(y, y_hat[0])
+
+    dataset = TabularDataset(data_path, batch_size=batch_size, task='regression')
+    runner = OrtRunner(model_path)
+    
+    return run_model(run_single_pass, runner, dataset, batch_size, num_runs, timeout)
+
+def run_sklearn_fp(model_path, batch_size, num_runs, timeout, data_path):
+    from utils.sklearn import SklearnRunner
+
+    def run_single_pass(sklearn_runner, dataset):
+        X, y = next(dataset)
+        y_hat = sklearn_runner.run(X)        
```

Comment:
Looks like code repetition from run.py file - how do run_optimized.py and run.py differ? Could the code be squeezed into a single file?